### ISSUE TITLE
torch backend: do("pad", ...) swaps (before, after) for asymmetric pad_width

### ISSUE THREAD

**Issue by Ryo-wtnb11** [jcmgray/autoray#35]
`do("pad", ...)` on the torch backend produces wrong results for any asymmetric `pad_width`: the translation to `torch.nn.functional.pad`'s flat format reverses not only the axis order (as torch requires) but also each `(before, after)` pair, so "pad before" becomes "pad after" and vice versa.

Because the two conventions coincide whenever `before == after`, every symmetric pad works fine — which is probably why this has gone unnoticed.

## Repro

```python
import numpy as np
import torch
import autoray as ar

x = np.arange(6.).reshape(2, 3)
pw = ((0, 1), (2, 0))  # 1 row after, 2 cols before

want = np.pad(x, pw)
got = ar.do("pad", torch.arange(6.).reshape(2, 3), pw)

print(np.array_equal(ar.to_numpy(got), want))  # False
```

```
expected                 torch backend got
[[0. 0. 0. 1. 2.]        [[0. 0. 0. 0. 0.]
 [0. 0. 3. 4. 5.]         [0. 1. 2. 0. 0.]
 [0. 0. 0. 0. 0.]]        [3. 4. 5. 0. 0.]]
```

Observed with autoray 0.10.1 (and the current `main`), torch 2.13.0, numpy 2.5.2. The numpy and jax backends agree with `np.pad`.

## Cause

In `torch_pad` (`autoray/autoray.py`, currently around line 4127):

```python
# numpy takes pads like ((0, 0), (1, 1), ... (n-1, n-1))
# torch takes pads like (n-1, n-1, n-2, n-2, n-3, n-3, ...)
pad = tuple(itertools.chain.from_iterable(pad_width))[::-1]
```

Flattening first and then reversing the flat tuple reverses the *pairs* too. Torch's format is `(before_last, after_last, before_secondlast, after_secondlast, ...)` — last axis first, but each pair still in `(before, after)` order.

## Fix

Reverse the axis order before flattening:

```python
pad = tuple(itertools.chain.from_iterable(reversed(pad_width)))
```

The single-tuple broadcast case (`len(pad) == 2 → pad * ndim`) is unaffected by this change and keeps working for the symmetric case; an asymmetric single tuple `((a, b),)` is also fixed by it, since the per-axis pair order is then correct.

Happy to send a PR if useful.

**Comment by jcmgray** [comment]
Thanks for finding this @Ryo-wtnb11, happy to accept a PR fixing!
